sources_list.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/python3
+
+import requests
+import os
+import yaml
+import json
+import itertools
+
+def req(url):
+    return requests.get(url, headers={"Accept": "application/vnd.github.raw"})
+
+def main():
+    if os.path.exists("iconset.json"):
+        with open("iconset.json", "r") as read_file:
+            iconset = json.load(read_file)
+    else:
+        iconset = req('https://raw.githubusercontent.com/maputnik/osm-liberty/gh-pages/iconset.json').json()
+    iconset_names = [list(group['svgs'].keys()) for group in iconset['iconGroups']]
+    iconset_names = [item for sublist in iconset_names for item in sublist]
+    # Remove the -11.svg and -15.svg from each name list to easily deduplicate
+    iconset_names = list(set([x.replace('_11.svg', '').replace('_15.svg', '') for x in iconset_names]))
+
+    if os.path.exists("omt.yaml"):
+        with open("omt.yaml", "r") as read_file:
+            omt_mapping = yaml.safe_load(read_file)
+    else:
+        r = req('https://api.github.com/repos/openmaptiles/openmaptiles/contents/layers/poi/mapping.yaml')
+        omt_mapping = yaml.safe_load(r.text)
+    omt_poi_class = omt_mapping['def_poi_mapping'].keys()
+    omt_poi_subclass = list(set(itertools.chain(*omt_mapping['def_poi_mapping'].values())))
+
+    if os.path.exists("maki.json"):
+        with open("maki.json", "r") as read_file:
+            maki_names = json.load(read_file)
+    else:
+        r = req('https://api.github.com/repos/mapbox/maki/contents/icons')
+        maki_names = r.json()
+    maki_names = [x['name'] for x in maki_names]
+    # Maki names are hyphenated; iconset names are underscored
+    maki_names = [x.replace('-', '_') for x in maki_names]
+    maki_names = [x.replace('.svg', '') for x in maki_names]
+
+    omt_unused = set(omt_poi_class).difference(iconset_names)
+    omt_missing = set(iconset_names).difference(omt_poi_class)
```

Comment:
```suggestion
    omt_missing = set(iconset_names).difference(omt_poi_class).difference(omt_poi_subclass)   
```

`subclass` could be used for styling as well, so they are not missing in OMT.

Also there are some keys which are both `class` and `subclass`, e.g. _bar_ -> https://github.com/openmaptiles/openmaptiles/blob/dfd20c647c1a20e45e5ac869d3769289db17979b/layers/poi/poi.yaml#L71

What do you think?

Reply:
I thought I had just scratched the surface of the problem, so I pushed the script in this state as a base for future experimentations.

OSM Liberty style uses icons very basically: it's just `"icon-image": "{class}_11"`.
In the future the subclass could be used, yes. But how?

- specify an icon for each subclass (or maybe for each class-subclass pair)
    => are there some features with an empty subclass? if so, do we need a fallback?
    => will we accept to duplicate lots of icons? or can we create aliases in sprite.json?
- for each subclass (or class-subclass pair), create rules in the style that select an icon

...what are your thoughts?

Additional reflections:

- Other OMT layers might have point features worthy of icons (aero things for instance)
- We don't use any of the `_15` sprites!